### PR TITLE
test_shed_too_large_request fix: disable compression

### PR DESCRIPTION
**Pull request by gusev-p**
The test relies on exact request size, this doesn't work if compression is applied. The driver enables compression only if both the server and the client agree on the codec to use. If compression package
(e.g. lz4) is not installed, the compression
is not used.

The trick with locally_supported_compressions is needed since I couldn't find any standard means to disable compression other than the compression flag
on the cluster object, which seemed too broad.

fixes: #12836

### CODE REVIEW

**Review by avikivity — APPROVED** [review]

**Review by nyh — APPROVED** [review]
Funky monkey-patching, but I like it ;-)
But I wonder - is there really no way to disable compression just for the session (which new_cql() creates anyway) without monkey-patching?

**Review by xemul — APPROVED** [review]
Test now passes :+1:

**Review by alecco — APPROVED** [review]

**Comment by alecco** [comment]
Perhaps add "Fixes https://github.com/scylladb/scylladb/issues/12836" ?

**Comment by gusev-p** [comment]
> Funky monkey-patching, but I like it ;-)
> But I wonder - is there really no way to disable compression just for the session (which new_cql() creates anyway) without monkey-patching?

Compression negotiation happens in the [Connection](https://github.com/scylladb/python-driver/blob/master/cassandra/connection.py#L1353), `compression` flag is passed from `Connection` constructor, `Connection` objects are created through [connection_factory](https://github.com/scylladb/python-driver/blob/master/cassandra/cluster.py#L1681) `Cluster` method, which takes it's default value from the Cluster's `compression` field. `connection_factory` is called from `pool.py` and it doesn't pass the value for compression parameter, so default value from `Cluster` `compression` field is used always.

**Comment by gusev-p** [comment]
v2:
  * "Fixes #12836" added;
  * A shorter string for testing;
  * Comments clarified.

File: test/cql-pytest/test_shedding.py [hunk 1]
```diff
@@ -69,8 +67,10 @@ def get_protocol_errors():
 
     # disable compression since we rely on specific request size
     with disable_compression():
-        # separate session is needed due to the cql_test_connection fixture,
-        # which checks that the session is not broken at the end of the test
+        # Big request causes the entire connection to be closed (see #8800 for the reasons).
+        # For some reason the driver doesn't try to reopen the connection to the same host,
+        # so we end up with a broken session and cql_test_connection fixture fails the test.
+        # We create a separate session to make the test work.
```
Comment (nyh) [inline test/cql-pytest/test_shedding.py:73]: Next time, please conflate such a patch that fixes a comment with the earlier patch that introduces the comment in the first place. But it's not important enough to require a new version. Let's just wait for the CI and merge this version.

**Review by nyh — APPROVED** [review]

File: test/cql-pytest/test_shedding.py (hunk 1 above)
Comment (gusev-p) [reply test/cql-pytest/test_shedding.py:73]: What does 'conflate' mean here? The original patch is already merged, we can't change it.

Comment (nyh) [reply test/cql-pytest/test_shedding.py:73]: Sorry, I was under the mistaken impression that this specific comment was introduced in the first patch in this series. But now I see I was wrong.

**Comment by gusev-p** [comment]
debug build failed, and it reproduces locally, looking into it

**Comment by gusev-p** [comment]
v3:
  * fixed debug builds, read memory limit from Scylla metrics.

**Comment by gusev-p** [comment]
ping

**Pull request merged**